### PR TITLE
Nav Redesign: Fix inconsistent submenu colors

### PR DESCRIPTION
**Pull request by okmttdhr**
Fix https://github.com/Automattic/dotcom-forge/issues/6801
Related https://github.com/Automattic/wp-calypso/pull/90136

## Proposed Changes

This PR fixes the inconsistent submenu highlight colors. 

| before | after |
|--------|--------|
| Screenshot 2024-05-02 at 16 26 58 | Screenshot 2024-05-02 at 16 26 26 | 
|(hover) Screenshot 2024-05-02 at 16 27 41| (hover) Screenshot 2024-05-02 at 16 27 51|

## Testing Instructions

* Prepare an Atomin site with Classic View 
* Go to the site's `_cli` and run `wp option update wpcom_classic_early_release 1`.
* Compare the submenu highlight colors in wp-admin (e.g., `/wp-admin/profile.php`) and Calypso page (e.g., `/home/<your-site>`)

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/calypso-color-schemes/src/shared/color-schemes/_blue.scss [hunk 1]
```diff
@@ -144,6 +144,12 @@ Used studio-blue for the primary+accent.
 	--color-sidebar-submenu-hover-background: transparent;
 	--color-sidebar-submenu-hover-text: var(--theme-text-color);
 
+	/* Sidebar Submenu - Nav Redesign */
+	--color-navredesign-sidebar-submenu-text: #e2ecf1; /* $menu-submenu-text: */
+	--color-navredesign-sidebar-menu-selected-text: var(--theme-text-color);
+	--color-navredesign-sidebar-submenu-selected-text: var(--theme-text-color);
+	--color-navredesign-sidebar-submenu-hover-text: var(--theme-text-color);
```
Comment (okmttdhr) [inline packages/calypso-color-schemes/src/shared/color-schemes/_blue.scss:151]: badge

The files located at `packages/calypso-color-schemes/src/shared/color-schemes/_*.scss` were initially added for Nav Unification and are tailored specifically to it, incorporating Calypso-specific color designs. However, given that Nav Redesign fundamentally diverges from Nav Unification, inconsistencies in code have arisen (to address inconsistencies in UI), particularly with the variable names used in the color schemes.

Therefore, in the future, I suggest that we import all Core color schemes as faithfully as possible to their original specifications. This includes maintaining the original variable names and color palette designs from Core. We better minimize using Calypso-specific names and designs to align our project more closely with Core standards.

**Comment by taipeicoder** [comment]
Added `Reviewer can merge` since @okmttdhr will be AFK until next Tuesday.

**Review by fushar — COMMENTED** [review]
I see the Default scheme (slug = `fresh`) still has inconsistent selected parent menu color:

|Calypso|wp-admin|
|-|-|
|image|image|

**Review by davemart-in — APPROVED** [review]
Tested. This works as expected. Nicely done!

**Comment by miksansegundo** [comment]
> I see the Default scheme (slug = fresh) still has inconsistent selected parent menu color:

I fixed that issue @fushar 

|wp-admin|Calypso BEFORE|Calypso AFTER|
|-|-|-|
|Screenshot 2567-05-02 at 22 00 38|Screenshot 2567-05-02 at 22 19 19|Screenshot 2567-05-02 at 22 20 37

I have also found another issue with that scheme: it does not change the text color of the selected menu on hover, while other schemes do it. In the following screenshot, I forced the hover using the dev tools.

|wp-admin|Calypso|
|-|-|
|Screenshot 2567-05-02 at 22 05 58|Screenshot 2567-05-02 at 22 37 25|

I think it's a better UX for users when the same Hover CSS logic is used for all schemes in Calypso. What do you think about this small inconsistency in the color of the hover effect?

I believe this PR is good to go 🚀

**Comment by fushar** [comment]
> I fixed that issue @fushar

Sweet, thanks!

> I have also found another issue with that scheme: it does not change the text color of the selected menu on hover, while other schemes do it. 

Interesting. Seems like a bug in Core side.

> I think it's a better UX for users when the same Hover CSS logic is used for all schemes in Calypso. What do you think about this small inconsistency in the color of the hover effect?

Agree. I think it's not a big deal; not worth an inconsistency to fix.

> I believe this PR is good to go 🚀

Thanks for taking care of this PR!

**Pull request merged**